src/transformers/models/auto/modeling_auto.py

```diff
@@ -847,6 +850,7 @@
 
 MODEL_FOR_VISUAL_QUESTION_ANSWERING_MAPPING_NAMES = OrderedDict(
     [
+        ("beit3", "Beit3ForVisualQuestionAnswering"),
```

Comment:
2 comments here:

- class needs to be renamed to `Beit3ForQuestionAnswering` for consistency with `ViltForQuestionAnswering`
- this does not seem correct. `Beit3ForQuestionAnswering` has its own architecture and it not a causal LM. This model needs to be added in utils/check_repo.py under the "non auto configured" models (unless it has the same API as ViLT)

Reply:
I have renamed the class, But the API both the classes looks very different.

Reply:
The model only needs to be added to this mapping in case the API is the same. What's their difference?

Could you try making the VQA pipeline work with `Beit3ForQuestionAnswering`?

Reply:
Done